rapid7_insight_agent/plugin.spec.yaml

```diff
@@ -0,0 +1,222 @@
+    primaryAddress:
+      title: "Primary Address"
+      type: primaryAddress
+      description: "Primary address"
+      required: false
+    uniqueIdentity:
+      title: "Unique Identity"
+      type: "[]uniqueIdentity"
+      description: "Unique identity"
+      required: false
+    vendor:
+      title: "Vendor"
+      type: string
+      description: "Vendor"
+      required: false
+    version:
+      title: "Version"
+      type: string
+      description: "Version"
+      required: false
+
+connection:
+  api_key:
+    title: API Key
+    description: User or Organization Key from the Insight Platform
+    type: credential_secret_key
+    example: a5zy0a6g-504e-46bz-84xx-1b3f5ci36l99
+    required: true
+  org_id:
+    title: Organization ID
+    description: Organization ID
+    type: credential_secret_key
+    example: 2cg99z6y-795n-7bzf-hj67-12355h554974
+    required: true
+  region:
+    title: Region
+    description: Region
+    type: string
+    default: United States
+    example: United States
+    required: true
+    enum:
+      - United States
+      - Europe
+      - Canada
+      - Australia
+      - Japan
+actions:
+  get_agent_details:
+    title: Get Agent Details
+    description: This action is used to find and display detailed information about a device
+    input:
+      agent:
+        title: Agent
+        description: IP address, MAC address, or hostname of the device to get information from
+        type: string
+        required: true
+        example: MaliciousHost
+    output:
+      agent:
+        title: Agent
+        description: Agent information
+        type: agent
+        required: true
+  quarantine:
+    title: Quarantine
+    description: Quarantine or unquarantine on a device
+    input:
+      agent_id:
+        title: Agent ID
+        description: The ID of the agent on the device to quarantine
+        type: string
+        example: a1cfb273c8e7d46a9e2a0e2dae01a0ce
+        required: true
+        order: 1
+      quarantine_state:
+        title: Quarantine State
+        description: Set to true to quarantine a host, set to false to unquarantine
+        type: boolean
+        default: true
+        example: true
+        required: true
+        order: 2
+      interval:
+        title: Interval
+        description: Length of time in seconds to try to take action on a device. This is also called Advertisement Period
+        type: int
+        default: 604800
+        example: 604800
+        required: true
+    output:
+      success:
+        title: Success
+        description: Was operation successful
+        type: boolean
+        required: true
+  check_agent_status:
+    title: Check Agent Status
+    description: Get the online status and quarantine state of an agent
+    input:
+      agent_id:
+        title: Agent ID
+        description: The ID of the agent on the device to get the status from
+        type: string
+        example: a1cfb273c8e7d46a9e2a0e2dae01a0ce
```

Comment:
Do we support both formats of the Agent ID? if not, we should support the version with `-`'s too.  Same in the other actions.